### PR TITLE
`secureheaders` recipe: add `Content-Security-Policy` header and `includeSubDomains; preload` to `Strict-Transport-Security`

### PR DESCRIPTION
**Pull request by apexkoder**
Related: #2019

## Bug Checklist

- [ ] Integration test added (see [test docs](https://blitzjs.com/docs/contributing#running-tests) if needed)

## Feature Checklist

- [ ] Integration test added (see [test docs](https://blitzjs.com/docs/contributing#running-tests) if needed)
- [ ] Documentation added/updated (submit PR to [blitzjs.com repo `canary` branch](https://github.com/blitz-js/blitzjs.com/tree/canary))

### CODE REVIEW

File: recipes/secureheaders/templates/core/secureheaders.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export const cspHashOf = (text: string) => {
 export const computeCsp = (props: Readonly<DocumentProps>) => {
   const scriptHash = cspHashOf(BlitzScript.getInlineScriptSource(props))
 
-  return `default-src 'self'; style-src 'self' 'unsafe-inline' fonts.googleapis.com; font-src 'self' data: fonts.gstatic.com; script-src 'self' ${
-    process.env.NODE_ENV === "production" ? "" : "'unsafe-eval'"
-  } ${scriptHash}`
+  return `default-src 'self'; img-src 'self' res.cloudinary.com data:; style-src 'self' 'unsafe-inline' fonts.googleapis.com; font-src 'self' data: fonts.gstatic.com; script-src 'self' ${
```
Comment (MrLeebo) [inline recipes/secureheaders/templates/core/secureheaders.ts:13]: Is this addition necessary? Not every app requires cloudinary, and allowing the data: protocol gives a malicious user a method to inject XSS attacks https://www.w3.org/TR/CSP3/#csp-directives

**Review by MrLeebo — CHANGES_REQUESTED** [review]

File: recipes/secureheaders/templates/core/secureheaders.ts (hunk 1 above)
Comment (PaulEbose) [reply recipes/secureheaders/templates/core/secureheaders.ts:13]: > Not every app requires cloudinary

It can be removed. I put in there as an example, since also not every app require Google fonts.

> allowing the data: protocol gives a malicious user a method to inject XSS attacks https://www.w3.org/TR/CSP3/#csp-directives

I'm adding `data:` protocol to the `img-src` directive. From the link you sent:

```md
To mitigate the risk of cross-site scripting attacks, web developers SHOULD include directives that regulate sources of script and plugins. They can do so by including:

• Both the [script-src](https://www.w3.org/TR/CSP3/#script-src) and [object-src](https://www.w3.org/TR/CSP3/#object-src) directives, or
• a [default-src](https://www.w3.org/TR/CSP3/#default-src) directive
```

This [web.dev](https://web.dev/csp-xss/#csp-targets-xss) article hints that the `script-src`, `object-src`, and `base-uri` directives are relevant for XSS

Comment (MrLeebo) [reply recipes/secureheaders/templates/core/secureheaders.ts:13]: I'm fine with keeping the data: protocol for img-src then, but I think cloudinary can be left out. Unlike cloudinary, Google fonts is used by the default app template and the secureheaders recipe should generate a policy that doesn't produce errors on a fresh blitz app.

It would be a good idea to add a comment next to that line letting users know they can remove the Google font if they've replaced the default CSS though.

**Review by MrLeebo — APPROVED** [review]
Great work on this 👍

**Review by Nisarga-Developer — APPROVED** [review]

**Review by beerose — APPROVED** [review]
Thanks for this PR!

**Comment by siddhsuresh** [comment]
Closing since required headers added in #3813
